tensorflow_cloud/gcp.py

```diff
@@ -0,0 +1,297 @@
+# limitations under the License.
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import google.auth
+
+
+def get_project_name():
+    # https://google-auth.readthedocs.io/en/latest/reference/google.auth.html
+    _, project_id = google.auth.default()
+    if project_id is None:
+        raise Exception('Could not determine the GCP project id.')
+
+    return project_id
+
+
+def validate_machine_configuration(
+        cpu_cores, memory, accelerator_type, accelerator_count):
+    valid_configurations = _get_valid_machine_configurations()
+    current_config = (
+        cpu_cores, memory, accelerator_type.value, accelerator_count)
+    if current_config not in valid_configurations:
+        raise ValueError(
+            'Invalid machine configuration: cpu_cores:{}, memory:{}, '
+            'accelerator_type:{}, accelerator_count:{}. Please see the '
+            'following AI platform comptibility table for all valid '
+            'configurations: '
+            'https://cloud.google.com/ml-engine/docs/using-gpus#'
+            'compute-engine-machine-types-with-gpu'.format(
+                cpu_cores, memory, str(accelerator_type), accelerator_count))
+
+
+def get_region():
+    return 'us-central1'
+
+
+def get_accelerator_type(accl_type):
+    if accl_type == 'CPU':
+        return 'ACCELERATOR_TYPE_UNSPECIFIED'
+    if accl_type == 'K80':
+        return 'NVIDIA_TESLA_K80'
+    if accl_type == 'P100':
+        return 'NVIDIA_TESLA_P100'
+    if accl_type == 'V100':
+        return 'NVIDIA_TESLA_V100'
+    if accl_type == 'P4':
+        return 'NVIDIA_TESLA_P4'
+    if accl_type == 'T4':
+        return 'NVIDIA_TESLA_T4'
+    else:
+        raise ValueError('Invalid accelerator type.')
+
+
+def get_machine_type(cpu_cores, memory):
+    config = (cpu_cores, memory)
+    if config == (4, 15):
+        return 'n1-standard-4'
+    if config == (8, 30):
+        return 'n1-standard-8'
+    if config == (16, 60):
+        return 'n1-standard-16'
+    if config == (32, 120):
+        return 'n1-standard-32'
+    if config == (64, 240):
+        return 'n1-standard-64'
+    if config == (96, 360):
+        return 'n1-standard-96'
+    if config == (2, 13):
+        return 'n1-highmem-2'
+    if config == (4, 26):
+        return 'n1-highmem-4'
+    if config == (8, 52):
+        return 'n1-highmem-8'
+    if config == (16, 104):
+        return 'n1-highmem-16'
+    if config == (32, 208):
+        return 'n1-highmem-32'
+    if config == (64, 416):
+        return 'n1-highmem-64'
+    if config == (96, 624):
+        return 'n1-highmem-96'
+    if config == (16, 14.4):
+        return 'n1-highcpu-16'
+    if config == (32, 28.8):
+        return 'n1-highcpu-32'
+    if config == (64, 57.6):
+        return 'n1-highcpu-64'
+    if config == (96, 86.4):
+        return 'n1-highcpu-96'
+    else:
+        raise ValueError('Invalid machine type.')
```

Comment:
This should 1) print what the user passed, 2) offer the list of valid options (e.g. via a link to docs, since the actual list is long)

Reply:
This actually should never happen, this is not a public API. 'Validate_machine_configurations' API above validates machine type at creation time and raises error if the input is invalid. That also provides link to the valid machine types list.